src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs

```diff
@@ -332,7 +332,7 @@ private static IPAddress ParseIPv6HexString(string hexAddress, bool isNetworkOrd
         {
             Debug.Assert(hexAddress.Length == 32);
             byte[] addressBytes = new byte[16];
-            if (isNetworkOrder)
+            if (isNetworkOrder || !BitConverter.IsLittleEndian)
```

Comment:
Is there a case where `isNetworkOrder` is not `!BitConverter.IsLittleEndian`?

Reply:
isNetworkOrder is a flag that the caller sets to indicate whether the source string is in network order as opposed to host order; this is a property of the particular /proc file (or other string) that is being read.  If the source string is in network order, it can be parsed directly, otherwise it needs to converted from host order to network order.

On a little-endian system, that conversion involves byte swapping.  However, on a big-endian system, host order **is** network order, so the conversion is a no-op, and the string can just be directly parsed either way.